lib/idempotency.rb

```diff
@@ -137,4 +148,42 @@ def unquote(str)
       str
     end
   end
+
+  # rubocop:disable Metrics/AbcSize
+  def with_apm_instrumentation(name, tags = {}, &block)
+    # Build nested instrumentation layers from innermost to outermost
+    instrumented_block = block
+
+    # Wrap with Sentry if enabled
+    if config.observability.sentry_enabled && defined?(Sentry)
```

Comment:
Could we do the Sentry check at load time? Like when `sentry_enabled` is turned on 🤔